tests/python/openassetio/managerAPI/test_host.py

```diff
@@ -41,20 +41,22 @@ def a_context():
 #   assessing when this is ported to cpp
 
 class Test_Host_init:
+
+    def test_when_inheriting_then_raises_TypeError(self):
+        with pytest.raises(TypeError) as err:
+            class _Derived(Host):
+                pass
+
+        assert str(err.value) == (
+            "type 'openassetio._openassetio.managerAPI.Host' is not an acceptable base type")
+
     def test_when_invalid_interface_then_raises_TypeError(self):
         with pytest.raises(TypeError) as err:
             Host(object())
 
         assert str(err.value).startswith("__init__(): incompatible constructor arguments")
 
 
-class Test_Host_interface:
-    def test(self, mock_host_interface):
-        # pylint: disable=protected-access
-        a_host = Host(mock_host_interface)
-        assert a_host._interface() is mock_host_interface
```

Comment:
[consider] Do we still want some kind of unit test for this? We could set some test-custom return value for the identifier a method of our mock, call that method and check its returned by `host.identifier()`?

Reply:
There is already `Test_Host_identifier` (and other methods) that check the call is forwarded to the provided `HostInterface`.

Reply:
Ahh perfect.